# Metadata Gladstone Exploratory Data Analysis

In [1]:
import pandas as pd

In [2]:
pwd

'/Users/gbaldonado/Developer/kenfong_ecoli_ml/Kenfong_Ecoli_ML/kenfong2024/dataset_folder/gladstone_2021'

In [3]:
gladstone_df = pd.read_csv("./metadata_gladstone_2021.csv")
gladstone_df.shape

(3254, 81)

In [4]:
gladstone_df.head()

Id Run accession  popPUNK_cluster__autocolour  ST__autocolour  \
0  30134_6_100    ERR4034085                            1              73   
1  30134_6_101    ERR4034086                            8             127   
2  30134_6_103    ERR4034087                            3              69   
3  30134_6_104    ERR4034088                            1              73   
4  30134_6_105    ERR4034089                            5             537   

  CC131_clades__autocolour Duplicate__autocolour  year     Lab  \
0                      NaN                   NaN  2004  Tromsø   
1                      NaN                   NaN  2015  Tromsø   
2                      NaN                   NaN  2013  Tromsø   
3                      NaN                   NaN  2005  Tromsø   
4                      NaN                   NaN  2011  Tromsø   

                      Hospital name Regional health authority  ...  \
0  Universitetssykehuset Nord-Norge                Helse Nord  ...   
1  Universitetssykehuset Nord-Norge                Helse Nord  ...   
2  Universitetssykehuset Nord-Norge                Helse Nord  ...   
3  Universitetssykehuset Nord-Norge                Helse Nord  ...   
4  Universitetssykehuset Nord-Norge                Helse Nord  ...   

  Ciprofloxacin_SIR__colour  ESBL__colour  Gentamicin_SIR__colour  \
0                   #57C4AD           NaN                 #57C4AD   
1                   #57C4AD       #57C4AD                 #57C4AD   
2                   #57C4AD           NaN                 #57C4AD   
3                   #57C4AD           NaN                 #57C4AD   
4                   #57C4AD           NaN                 #57C4AD   

  Meropenem_SIR__colour Nalidixin_SIR__colour  \
0               #57C4AD                   NaN   
1               #57C4AD                   NaN   
2               #57C4AD                   NaN   
3               #57C4AD                   NaN   
4               #57C4AD                   NaN   

  Piperacillin_tazobactam_SIR__colour Sulfa_SIR__colour  \
0                             #57C4AD               NaN   
1                             #EDA247               NaN   
2                             #57C4AD               NaN   
3                             #57C4AD               NaN   
4                             #57C4AD               NaN   

  Tigecyklin_SIR__colour Tobramycin_SIR__colour Trimetoprim_sulfa_SIR__colour  
0                    NaN                    NaN                       #57C4AD  
1                #57C4AD                    NaN                       #57C4AD  
2                #57C4AD                    NaN                       #DB4325  
3                    NaN                    NaN                       #57C4AD  
4                #57C4AD                    NaN                       #57C4AD  

[5 rows x 81 columns]

In [12]:
gladstone_df["Ampicillin_SIR"]

0       S
1       R
2       R
3       S
4       S
       ..
3249    S
3250    S
3251    S
3252    R
3253    S
Name: Ampicillin_SIR, Length: 3254, dtype: object

In [6]:
gladstone_df["Ampicillin_SIR__colour"]

0       #57C4AD
1       #DB4325
2       #DB4325
3       #57C4AD
4       #57C4AD
         ...   
3249    #57C4AD
3250    #57C4AD
3251    #57C4AD
3252    #DB4325
3253    #57C4AD
Name: Ampicillin_SIR__colour, Length: 3254, dtype: object

In [7]:
# Count NaN values per column
# pd.set_option('display.max_rows', None)
# pd.set_option('display.max_columns', None)

nan_counts = gladstone_df.isna().sum()
print(nan_counts)

Id                                        0
Run accession                             0
popPUNK_cluster__autocolour               0
ST__autocolour                            0
CC131_clades__autocolour               2976
                                       ... 
Piperacillin_tazobactam_SIR__colour       0
Sulfa_SIR__colour                      3254
Tigecyklin_SIR__colour                  767
Tobramycin_SIR__colour                 3058
Trimetoprim_sulfa_SIR__colour             0
Length: 81, dtype: int64


From the columns, it appears we have susceptibility, intermediate, resistance (SIR) data on

Ampicillin
Aztreonam
Cefotaxim
Ceftazidim
Cefuroxim
Ciprofloxacin
Gentamicin
Meropenem
Nalidixin
Piperacillin
Sulfa
Tigecyklin
Tobramycin
Trimethoprim

In [8]:
accession_list = gladstone_df["Run accession"].to_list()
print(len(accession_list))

3254


In [9]:
with open('accession_keys_gladstone.txt', 'w') as file:
    # Write each item in the list to the file
    for item in accession_list:
        file.write(item + '\n')

We have 3,254 accession numbers from the Gladstone paper.

In [10]:
gladstone_df = pd.read_excel("./Norway E. coli Resistance Study.xlsx")
gladstone_df.shape

(3254, 63)

In [11]:
gladstone_df

lane Run accession  popPUNK_cluster   ST CC131_clades Duplicate  \
0     30134_6#100    ERR4034085                1   73          NaN       NaN   
1     30134_6#101    ERR4034086                8  127          NaN       NaN   
2     30134_6#103    ERR4034087                3   69          NaN       NaN   
3     30134_6#104    ERR4034088                1   73          NaN       NaN   
4     30134_6#105    ERR4034089                5  537          NaN       NaN   
...           ...           ...              ...  ...          ...       ...   
3249   31663_7#85    ERR4037446                4  131            A       NaN   
3250   31663_7#89    ERR4037447                6   58          NaN       NaN   
3251    31663_7#9    ERR4037427                4  131           C1       NaN   
3252   31663_7#93    ERR4037448                6  101          NaN       NaN   
3253   31663_7#98    ERR4037449                9  141          NaN       NaN   

      year        Lab                     Hospital name  \
0     2004     Tromsø  Universitetssykehuset Nord-Norge   
1     2015     Tromsø  Universitetssykehuset Nord-Norge   
2     2013     Tromsø  Universitetssykehuset Nord-Norge   
3     2005     Tromsø  Universitetssykehuset Nord-Norge   
4     2011     Tromsø  Universitetssykehuset Nord-Norge   
...    ...        ...                               ...   
3249  2014       Ahus      Akershus universitetssykehus   
3250  2014       Ahus      Akershus universitetssykehus   
3251  2013  Haukeland     Haukeland universitetssykehus   
3252  2014       Ahus      Akershus universitetssykehus   
3253  2015       Ahus      Akershus universitetssykehus   

     Regional health authority  ... Ciprofloxacin_NS        ESBL  \
0                   Helse Nord  ...                S  -            
1                   Helse Nord  ...                S           -   
2                   Helse Nord  ...                S  -            
3                   Helse Nord  ...                S  -            
4                   Helse Nord  ...                S  -            
...                        ...  ...              ...         ...   
3249             Helse Sør-Øst  ...                S  -            
3250             Helse Sør-Øst  ...                S  -            
3251                Helse Vest  ...                S         NaN   
3252             Helse Sør-Øst  ...                S  -            
3253             Helse Sør-Øst  ...                S  -            

     Gentamicin_SIR Meropenem_SIR Nalidixin_SIR Piperacillin_tazobactam_SIR  \
0                 S             S           NaN                           S   
1                 S             S           NaN                           I   
2                 S             S           NaN                           S   
3                 S             S           NaN                           S   
4                 S             S           NaN                           S   
...             ...           ...           ...                         ...   
3249              S             S           NaN                           S   
3250              I             S           NaN                           S   
3251              S             S             S                           S   
3252              S             S           NaN                           S   
3253              S             S           NaN                           S   

     Sulfa_SIR Tigecyklin_SIR Tobramycin_SIR Trimetoprim_sulfa_SIR  
0          NaN            NaN            NaN                     S  
1          NaN              S            NaN                     S  
2          NaN              S            NaN                     R  
3          NaN            NaN            NaN                     S  
4          NaN              S            NaN                     S  
...        ...            ...            ...                   ...  
3249       NaN            NaN            NaN                     S  
3250       NaN           